### PR TITLE
Add missing axis type to documentation parallelism example

### PR DESCRIPTION
**Pull request by RunnersNum40**
As mentioned and fixed for the test suite in https://github.com/patrick-kidger/equinox/pull/1146, [JAX 0.8.1](https://docs.jax.dev/en/latest/changelog.html#jax-0-8-1-november-18-2025) changed the default `axis_types` of `jax.make_mesh` to `jax.sharding.AxisType.Explicit`. This PR fixes the docs as well, which won't run with an up-to-date version of JAX.

### CODE REVIEW

**Comment by patrick-kidger** [comment]
Thank you for the fix! :)
